src/nla/cssp.rs

```diff
@@ -186,7 +186,7 @@ pub fn cssp_connect<S: Read + Write>(
         link.get_peer_certificate()?,
         "No public certificate available"
     )?
-    .to_der()?;
+    .0;
```

Comment:
What's this syntax here? I think I recall it vaguely from the Rust book, but can't find it and it's difficult to Google.

Reply:
https://doc.rust-lang.org/reference/expressions/tuple-expr.html

Reply:
Essentially, anonymous structs with non-named-fields

Reply:
Ah that's right, thank you. In this case, it's being used on a
```rust
pub struct Certificate(pub Vec<u8>);
```

but the operation results in `certificate_der: [u8]`. Do you understand how `Vec<u8>` is becoming `[u8]`?

Reply:
Yup, so `[u8]` refers to a *slice*, aka a contigous array of something. Rust has a special trait called `Deref` which allows the compiler to auto-convert a reference to one type into a reference to another. This is for example how `Arc<T>` works, it implements `Deref<Item = T>` which is how you can access fields on `T` from it. `Vec<T>` stores it's elements as a contigous array too, only it's dynamically sized and mutable in length, this allows `Vec<T>` to safely implement `Deref<Item = [u8]>` which ultimately returns `&[u8]`.